### PR TITLE
[tests] change 1.2 builds to 1.3

### PR DESCRIPTION
**Pull request by jwhui**

### CODE REVIEW

File: tests/scripts/thread-cert/thread_cert.py [hunk 1]
```diff
@@ -483,14 +483,19 @@ def _parse_params(self, params: Optional[dict]) -> dict:
 
         if params.get('is_bbr') or params.get('is_otbr'):
             # BBRs must use thread version 1.2
-            assert params.get('version', '1.2') == '1.2', params
-            params['version'] = '1.2'
+            version = params.get('version', '1.2')
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/thread_cert.py:486]:
```suggestion
            version = params.get('version', '1.3')
```
Should be 1.3 by default? 

or maybe we can define a const like `consts.LATEST_THREAD_VERSION` ?

File: tests/scripts/thread-cert/thread_cert.py [hunk 2]
```diff
@@ -483,14 +483,19 @@ def _parse_params(self, params: Optional[dict]) -> dict:
...
             params.setdefault('bbr_registration_jitter', config.DEFAULT_BBR_REGISTRATION_JITTER)
         elif params.get('is_host'):
             # Hosts must not specify thread version
             assert params.get('version', '') == '', params
             params['version'] = ''
 
+        # use 1.3 node for 1.2 tests
+        if 'version' in params and params['version'] == '1.2':
+            params['version'] = '1.3'
+
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/thread_cert.py:498]:
```suggestion
```

the code looks strange.

File: script/test [hunk 3]
```diff
@@ -559,10 +559,10 @@ envsetup()
         export RADIO_DEVICE="${OT_BUILDDIR}/openthread-simulation-${THREAD_VERSION}/examples/apps/ncp/ot-rcp"
         export OT_CLI_PATH="${OT_BUILDDIR}/openthread-posix-${THREAD_VERSION}/src/posix/ot-cli"
 
-        if [[ ${THREAD_VERSION} == "1.2" ]]; then
+        if [[ ${THREAD_VERSION} == "1.3" ]]; then
             export RADIO_DEVICE_1_1="${OT_BUILDDIR}/openthread-simulation-1.1/examples/apps/ncp/ot-rcp"
             export OT_CLI_PATH_1_1="${OT_BUILDDIR}/openthread-posix-1.1/src/posix/ot-cli"
-            export OT_CLI_PATH_1_2_BBR="${OT_BUILDDIR}/openthread-posix-1.2-bbr/src/posix/ot-cli"
+            export OT_CLI_PATH_1_3_BBR="${OT_BUILDDIR}/openthread-posix-1.3-bbr/src/posix/ot-cli"
```
Comment (simonlingoogle) [inline script/test:565]:
```suggestion
            export OT_CLI_PATH_BBR="${OT_BUILDDIR}/openthread-posix-1.3-bbr/src/posix/ot-cli"
```
nit: might be a large change.

File: script/test [hunk 4]
```diff
@@ -124,7 +124,7 @@ build_posix()
     local version="$1"
     local options=("-DOT_MESSAGE_USE_HEAP=ON" "-DOT_THREAD_VERSION=${version}" "-DBUILD_TESTING=ON")
 
-    if [[ ${version} == "1.2" ]]; then
+    if [[ ${version} == "1.3" ]]; then
```
Comment (simonlingoogle) [inline script/test:127]: use `!= "1.1"` ?

File: script/check-size [hunk 5]
```diff
@@ -151,9 +151,9 @@ size_nrf52840_version()
 
     local thread_version=$1
 
-    if [[ ${thread_version} == "1.2" ]]; then
+    if [[ ${thread_version} == "1.3" ]]; then
```
Comment (simonlingoogle) [inline script/check-size:154]:
```suggestion
    if [[ ${thread_version} != "1.1" ]]; then
```

**Review by simonlingoogle — COMMENTED** [review]
SGTM overall. Thanks. 

For some test scripts (e.g. script/test), should we make them work with both 1.2 and 1.3?
Might be useful if we want to do 1.2 and 1.3 co-op tests in the future.

File: tests/scripts/thread-cert/thread_cert.py (hunk 2 above)
Comment (jwhui) [reply tests/scripts/thread-cert/thread_cert.py:498]: This is to handle cases where `version` was set to `1.2` or the `version` key does not exist at all. Do you have a suggestion to improve this?

**Review by abtink — APPROVED** [review]
LGTM. 👍

File: tests/scripts/thread-cert/thread_cert.py [hunk 6]
```diff
@@ -482,15 +482,20 @@ def _parse_params(self, params: Optional[dict]) -> dict:
...
+            params['version'] = version
             params.setdefault('bbr_registration_jitter', config.DEFAULT_BBR_REGISTRATION_JITTER)
         elif params.get('is_host'):
             # Hosts must not specify thread version
             assert params.get('version', '') == '', params
             params['version'] = ''
 
+        # use 1.3 node for 1.2 tests
+        if params.get('version') == '1.2':
+            params['version'] = '1.3'
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/thread_cert.py:497]: Is it because we are not changing 'version': '1.2' in test scripts?

**Review by simonlingoogle — APPROVED** [review]
LGTM 👍 

with a small question

File: tests/scripts/thread-cert/thread_cert.py (hunk 6 above)
Comment (jwhui) [reply tests/scripts/thread-cert/thread_cert.py:497]: Yes. I wasn't sure if we should be updating them to 1.3 given that they are really testing 1.2 functionality.

Comment (simonlingoogle) [reply tests/scripts/thread-cert/thread_cert.py:497]: The `version` in test topology shows what the version of the device should be. 
- 1.1 means the device should be using 1.1 firmwares
- 1.2 means the device should be using 1.2 firmwares (and also 1.3 firmware as of now)
- `not specified` means the device should use 1.1/1.2/1.3 firmwares according to the current test environment

I think for most `1.2` versions, they actually mean `1.2` or `1.3`, whichever is specified by the environment variable `THREAD_VERSION`.

I think we can do it later in a future PR.

**Pull request merged**